internal/graph/lr2streams.go

```diff
@@ -261,7 +287,7 @@ func publishResultToParentStream(
 
 	// The cursor for the response is that of the parent response + the cursor from the result itself.
 	afterResponseCursor, err := combineCursors(
-		ci.responsePartialCursor(),
+		responseCursor,
 		result.AfterResponseCursor,
 	)
 	if err != nil {
```

Comment:
I can't comment on it, but:

```go
        mappedResource, err := foundResources.mapPossibleResource(result.Resource)
        ...
	missingContextParameters := mapz.NewSet(mappedResource.MissingContextParams...)
	missingContextParameters.Extend(result.Resource.MissingContextParams)
	missingContextParameters.Extend(additionalMissingContext)
	mappedResource.MissingContextParams = missingContextParameters.AsSlice()
```
Why not just have `mappedResource.MissingContextParams` be a set so that you can just do:
```go
        mappedResource, err := foundResources.mapPossibleResource(result.Resource)
        mappedResource.MissingContextParams.Extend(result.Resource.MissingContextParams)
        mappedResource.MissingContextParams.Extend(additionalMissingContext)
```
and then convert to a slice when you need to?

Reply:
Its a proto so it can't be a set